src/main/cocoa/decode.mm

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2025 Linux Studio Plugins Project <https://lsp-plug.in/>
+ *           (C) 2025 Vladimir Sadovnikov <sadko4u@gmail.com>
+ *           (C) 2025 Marvin Edeler <marvin.edeler@gmail.com>
+ *
+ * This file is part of lsp-ws-lib
+ * Created on: 16 June 2025
+ *
+ * lsp-ws-lib is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * any later version.
+ *
+ * lsp-ws-lib is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with lsp-ws-lib. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <lsp-plug.in/common/types.h>
+
+#ifdef PLATFORM_MACOSX
+
+#include <lsp-plug.in/ws/cocoa/decode.h>
+#include <Cocoa/Cocoa.h>
+
+namespace lsp
+{
+    namespace ws
+    {
+        namespace cocoa
+        {
+            mcb_t decode_mcb(NSEvent* event) 
+            {
+                NSUInteger button = [event buttonNumber]; // Get mouse button index
+                if (button <= 6) {
+                    return static_cast<mcb_t>(MCB_LEFT + button);  // Direct mapping
+                }
+                return MCB_NONE;
+            }
+
+            //TODO: do we need to map scroll direction set by user? macos scrolls inverted by default.
```

Comment:
If we want to handle scroll direction same for all systems, then I suppose it is a good solution to detect the true scroll direction instead of what MacOS gives to us. At least we can make a conditional compilation here and check which solution better fits our needs.